ratis-server/src/main/java/org/apache/ratis/server/impl/VoteContext.java

```diff
@@ -131,6 +132,9 @@ RaftPeer recognizeCandidate(long candidateTerm) {
    * See Section 5.4.1 Election restriction
    */
   boolean decideVote(RaftPeer candidate, TermIndex candidateLastEntry) {
+    if (impl.getRole().getCurrentRole() == RaftProtos.RaftPeerRole.LISTENER) {
+      return reject("The server " + impl.getId() + " is listener and should not participate in elections");
```

Comment:
@codings-dan , thanks a lot for working on this!  The id is already printed inside `reject(..)`.  How about changing the message to the following?
```java
 return reject("this server is a listener, who is a non-voting member");
```

Reply:
Good catch! I have change the message.